src/parser/solar.rs

```diff
@@ -215,303 +205,273 @@ impl<'ast> Visit<'ast> for LintspecVisitor<'ast> {
-                span_to_text_range(span, visitor.source_map)
-            } else {
-                span_to_text_range(item.span, visitor.source_map)
-            };
-
-            let natspec = extracted_natspec.map(|(ns, _)| ns.populate_returns(&returns));
-
-            let def = if item_function.kind.is_constructor() {
-                Definition::Constructor(ConstructorDefinition {
-                    parent,
-                    span,
-                    params,
-                    natspec,
-                })
-            } else if item_function.kind.is_modifier() {
-                Definition::Modifier(ModifierDefinition {
-                    parent,
-                    span,
-                    params,
-                    natspec,
-                    name: item_function.header.name.unwrap().to_string(),
-                    attributes: Attributes {
-                        visibility: item_function.header.visibility.into(),
-                        r#override: item_function.header.override_.is_some(),
-                    },
-                })
-            } else if item_function.kind.is_function() {
-                Definition::Function(FunctionDefinition {
-                    parent,
-                    name: item_function.header.name.unwrap().to_string(),
-                    returns: returns.clone(),
-                    attributes: Attributes {
-                        visibility: item_function.header.visibility.into(),
-                        r#override: item_function.header.override_.is_some(),
-                    },
-                    span,
-                    params,
-                    natspec,
-                })
-            } else {
-                return None;
-            };
-
-            Some(def)
-        } else {
-            None
-        }
-    }
-}
-
-impl<'ast> Extract<'ast> for solar_parse::ast::VariableDefinition<'ast> {
-    fn extract_definition(
-        visitor: &mut LintspecVisitor<'ast>,
-        item: &'ast Item<'ast>,
-    ) -> Option<Definition> {
+impl Extract for &solar_parse::ast::ItemFunction<'_> {
+    fn extract_definition(self, item: &Item, visitor: &mut LintspecVisitor) -> Option<Definition> {
         let parent = visitor.current_parent.last().cloned();
+        let params =
+            variable_definitions_to_identifiers(self.header.parameters, visitor.source_map);
 
-        if let ItemKind::Variable(item_variable) = &item.kind {
-            let extracted_natspec =
-                extract_natspec(&item.docs, visitor.source_map, parent.as_ref()).unwrap_or(None);
-
-            let span = if let Some((_, span)) = extracted_natspec {
-                span_to_text_range(span, visitor.source_map)
-            } else {
-                span_to_text_range(item.span, visitor.source_map)
-            };
+        let extracted_natspec =
+            extract_natspec(&item.docs, visitor.source_map, parent.as_ref()).unwrap_or(None);
```

Comment:
Just realised we don't really handle errors happening extract_natspec (it returns a result option). On the other hand, I now realise the path to have an error not caught previously is not that clear  for me